code/modules/clothing/shoes/cowboy.dm

```diff
@@ -15,6 +15,10 @@
 		var/mob/living/simple_animal/hostile/retaliate/snake/bootsnake = new/mob/living/simple_animal/hostile/retaliate/snake(src)
 		occupants += bootsnake
 
+/obj/item/clothing/shoes/cowboy/Destroy()
+	for(var/mob/living/occupant in occupants)
+		occupant.forceMove(loc)
```

Comment:
Assertion: The snake is already inside our contents, and thus will already be qdeleted on destroy.
So the issue here is the occupants list failing to update. 
You should use Exited to remove the snake from the list when it leaves, that way there's no potential gottchas for deleting the snake or something